win32ss/user/ntuser/menu.c

```diff
@@ -3023,7 +3023,14 @@ static BOOL FASTCALL MENU_ShowPopup(PWND pwndOwner, PMENU menu, UINT id, UINT fl
         if ((y - height) < monitor->rcMonitor.top || y >= monitor->rcMonitor.bottom)
             y = monitor->rcMonitor.bottom - height;
         else
-            y -= height;
+        {
+            INT adjHgt = y + UserGetSystemMetrics(SM_CYMENUSIZE) +
```

Comment:
This should probably not be a fixed number, unless it is absolutely impossible to have a multiline menu item?

Reply:
Is this same mechanism valid both for the 'first' menu, and any 'sub' menu's?

Reply:
(I am unsure about the answer to both questions btw, experiments could probably answer this).

Now that I think about it, I used to have a test application for this, I'll try to find that.

Reply:
> Is this same mechanism valid both for the 'first' menu, and any 'sub' menu's?

This question seems already answered by my test-video https://jira.reactos.org/secure/attachment/68344/68344_menu_pos_fix-05_isReallyGood.webm
where I tested several submenus in the same popup menu. Seems that the implementation can handle that well.

Reply:
I tried to make a two line menu by editing your MenuTest program, but I was not able to create a menu entry with more than one line. Also, I searched the Internet and could not find even a single example of a menu with two lines. Therefore I believe that this is not possible. Here is an example of one of my tests.

![Test_for_2-line_menu](https://github.com/reactos/reactos/assets/32620577/48817563-683d-4a81-9c52-6eae6fe8d97a)

Reply:
> I tried to make a two line menu by editing your MenuTest program, but I was not able to create a menu entry with more than one line

Having a `\n` in the middle of the long string doesn't create a second line I guess?

Reply:
@HBelusca, I tested what you suggested and the backslash character just shows up like any other as you can see here.

![Test_for_2-line_menu-02](https://github.com/reactos/reactos/assets/32620577/d410cf12-3bc9-4793-84d1-f1f67b622e9e)